### PR TITLE
Meshoptimizer Update: Restrict edge collapses to avoid triangle flipping

### PR DESCRIPTION
**Pull request by fire**
@reduz 

- [x] Feel free to change my pr to put it in the proper patch format.
- [x] Marked as draft because it has not finished the proper thirdparty work.
- [x] Missing proper attribution in the commit message.

https://github.com/zeux/meshoptimizer/commit/e4e43fe36e7a8705e602e7ca2f9fb795ded1d0b9

Fixes only one out of the two of: https://github.com/godotengine/godot/issues/44551

### CODE REVIEW

**Comment by fire** [comment]
[AUTOLOD+Project.zip](https://github.com/godotengine/godot/files/5745592/AUTOLOD%2BProject.zip)

Test project.

image

**Comment by fire** [comment]
Needs an update since https://github.com/zeux/meshoptimizer/pull/211 is merged.

**Review by reduz — APPROVED** [review]

**Comment by akien-mga** [comment]
So what's the actual provenance of this code? Does it still have Godot specific modifications? Are those still needed after upstream changes?

**Comment by fire** [comment]
It still has mods. I don’t know how to to generate a patch.

On Dec 30, 2020, at 12:36 AM, Rémi Verschelde <notifications@github.com> wrote:

﻿
So what's the actual provenance of this code? Does it still have Godot specific modifications? Are those still needed after upstream changes?

**Comment by akien-mga** [comment]
- Backup the file.
- Copy upstream code in place of our version, `git add` it.
- Copy backed up modified file in place of upstream version, `git diff > patch`.

**Comment by fire** [comment]
@akien-mga The upstream updated and all mods were removed.

File: thirdparty/README.md [hunk 1]
```diff
@@ -339,18 +339,11 @@ File extracted from upstream release tarball:
 ## meshoptimizer
 
 - Upstream: https://github.com/zeux/meshoptimizer
-- Version: 0.15 (2020)
+- Version: git (e4e43fe36e7a8705e602e7ca2f9fb795ded1d0b9, 2020)
 - License: MIT
 
-File extracted from upstream release tarball:
-
```
Comment (akien-mga) [inline thirdparty/README.md:346]: You should keep this part.

Comment (fire) [reply thirdparty/README.md:346]: Changed to `File extracted from upstream tarball:`

File: thirdparty/README.md [hunk 2]
```diff
@@ -339,18 +339,12 @@ File extracted from upstream release tarball:
...
+- File extracted from upstream tarball:
```
Comment (akien-mga) [inline thirdparty/README.md:345]: Please keep it as it was...

Comment (akien-mga) [reply thirdparty/README.md:345]:
```suggestion
File extracted from upstream tarball:

```

**Comment by akien-mga** [comment]
Well let's merge I'll do documentation fixes at some point in the future.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**